### ISSUE TITLE
Unhandled TDS_ORDERBY token causes driver crash with ordered selects

### ISSUE THREAD

**Issue by driseley** [DataAction/AdoNetCore.AseClient#104]
We came across an issue where the driver was crashing when executing SQL with Order By statements.  The exceptions varied, but here is one example:

```   
System.InvalidOperationException : "Operation is not valid due to the current state of the object."

at AdoNetCore.AseClient.Internal.ReadablePartialStream.Read(Byte[] buffer, Int32 offset, Int32 count) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\ReadablePartialStream.cs:line 44
   at AdoNetCore.AseClient.Internal.StreamReadExtensions.ReadInt(Stream stream) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\StreamReadExtensions.cs:line 31
   at AdoNetCore.AseClient.Internal.FormatItem.ReadTypeInfo(FormatItem format, Stream stream, Encoding enc) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\FormatItem.cs:line 137
   at AdoNetCore.AseClient.Internal.FormatItem.ReadForParameter(Stream stream, Encoding enc, TokenType srcTokenType) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\FormatItem.cs:line 128
   at AdoNetCore.AseClient.Token.ParameterFormatCommonToken.Read(Stream stream, DbEnvironment env, IFormatToken previousFormatToken) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Token\ParameterFormatCommonToken.cs:line 64
   at AdoNetCore.AseClient.Token.ParameterFormat2Token.Create(Stream stream, DbEnvironment env, IFormatToken previous) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Token\ParameterFormat2Token.cs:line 15
   at AdoNetCore.AseClient.Internal.TokenParser.ParseInternal(Stream stream, DbEnvironment env)+MoveNext() in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\TokenParser.cs:line 26
   at System.Collections.Generic.List`1.AddEnumerable(IEnumerable`1 enumerable)
   at AdoNetCore.AseClient.Internal.TokenParser.Parse(Stream stream, DbEnvironment env) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\TokenParser.cs:line 14
   at AdoNetCore.AseClient.Internal.RegularSocket.ReceiveTokens(DbEnvironment env) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\RegularSocket.cs:line 127
   at AdoNetCore.AseClient.Internal.InternalConnection.ReceiveTokens(ITokenHandler[] handlers) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\InternalConnection.cs:line 95
   at AdoNetCore.AseClient.Internal.InternalConnection.InternalExecuteAsync(AseCommand command, AseTransaction transaction, TaskCompletionSource`1 rowsAffectedSource, TaskCompletionSource`1 readerSource, CommandBehavior behavior) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\InternalConnection.cs:line 221
--- End of stack trace from previous location where exception was thrown ---
   at AdoNetCore.AseClient.Internal.InternalConnection.ExecuteReader(CommandBehavior behavior, AseCommand command, AseTransaction transaction) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\Internal\InternalConnection.cs:line 287
   at AdoNetCore.AseClient.AseCommand.ExecuteReader(CommandBehavior behavior) in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\AseCommand.cs:line 183
   at AdoNetCore.AseClient.AseCommand.ExecuteReader() in C:\git-repos\AdoNetCore.AseClient\src\AdoNetCore.AseClient\AseCommand.cs:line 197
   at SybaseDriverTest.Program.Main(String[] args) in C:\git-repos\AdoNetCore.AseClient\src\SybaseDriverTest\Program.cs:line 22```
```

The crash is reproducible using the the following sql:

```sql
select *,* from master..syscolumns where 1 = 2 order by 31, 32, 33
```

After building a local copy of the driver and enabling logging - the following was noticed:

```
!!! Hit unknown token type TDS_ORDERBY !!!
!!! Hit unknown token type 31 !!!
```

Looking at the TDS spec it states:

`TDS_ORDERBY, # Columns, column #`

`# Columns` : This is the number of columns in the order-by clause. This argument is a two-byte, unsigned integer.

Current the code handles this token in the `CatchAllToken` class, reading a single byte as the length field, corrupting the reading of the TDS stream ( which causes the varying crashes ).

I will submit a PR with a test and proposed fix for this issue

**Comment by senseibaka** [comment]
Sections 5.2.1/5.2.2/5.2.3 of the spec talk about what the token's type indicates about the token's length. It looks like I completely misinterpreted it when implementing `CalculateRemainingLength` in the catch-all token.

Talk more in the PR

**Comment by driseley** [comment]
I have tested the fix for #107 locally and it addresses the issue we were having
